### DISCUSSION TITLE
PUDL Development Setup for Linux

### DISCUSSION THREAD

**Discussion by Nancy9ice** [catalyst-cooperative/pudl#3556] [Help Me!] [answered]
Hello, 

I've been trying to set up PUDL on my linux and have been following the instructions but I keep getting errors when I use the command: **make install-pudl**. Attached is a screenshot that contains the error. The package name changes if I retry the command. And honestly the expected hash is what my package has as seen in the third image so I'm wondering why I'm getting this error.
![screenshot1](https://github.com/catalyst-cooperative/pudl/assets/103530451/babd9889-4f30-49f5-a121-83d91b1d8396)
![screenshot2](https://github.com/catalyst-cooperative/pudl/assets/103530451/2666bc71-7d8c-4c2c-91bd-fe1dc87feff8)
![screenshot3](https://github.com/catalyst-cooperative/pudl/assets/103530451/e49d6658-678c-4135-a1c9-a983b4be6dbc)

@aesharpe @zaneselvans

**Comment by zaneselvans** [comment]
Not sure what's going on here. Is it possible there's a corrupted package in your local cache? Could you try purging the cached packages and attempting to create the environment again?  Are you installing on the `main` branch of PUDL?

**Reply by Nancy9ice** [reply]
I just purged the cached packages and re-ran the command but it didn't work. And yes, I'm installing it on the main branch

**Reply by jdangerx** [reply]
What version of `mamba` are you running?

Are you able to run `mamba install gcsfs==2024.3.1` directly, or does that come up with the same hash mismatch?

**Reply by aesharpe** [reply]
She mentioned in our office hours call that she was using mamba version 1.5.8

**Comment by Nancy9ice** [comment, marked as answer]
Hello,

I apologize for the late response. It has finally worked! I had to delete the cloned repo and start all over again on the development setup and it just worked. 

Thank you for your help! And thank you @aesharpe for recommending that I start from scratch

**Comment by dataengineergaurav** [comment]
Hello @Nancy9ice 
I am facing similar error. I tried to delete repo, updated my forked repo and tried again but no success. I'm also using mamba 1.5.8

Can you share what commands did you use for resolve this? 
![Screenshot from 2024-05-04 15-18-04](https://github.com/catalyst-cooperative/pudl/assets/8761197/e96c9c60-4c80-4b14-91e7-4485def06925)
![Screenshot from 2024-05-04 15-18-36](https://github.com/catalyst-cooperative/pudl/assets/8761197/c8a8f932-a24f-4751-9277-3f389243fdf2)

![image](https://github.com/catalyst-cooperative/pudl/assets/8761197/ab1607f0-1147-4266-88e4-9c8e654d08f3)

**Comment by Nancy9ice** [comment]
So sorry about this. I understand how frustrating it can be. I'll advise you do the following and see if it works:
- Delete the folder from your linux (That is, PycharmProjects/pudl)
- Delete the PUDL repo from your github
- Fork the repo again
- Reclone the repo in your linux
- Confirm that your mamba and conda are still intact
- run 'make install-pudl'

These are the steps I took to make it work. I hope it works for you and will be waiting for your update on the outcome.

**Reply by dataengineergaurav** [reply]
Hi @Nancy9ice 

I have just 

1. Deleted pudl from my local machine. 
2. Deleted github forked repo
3. Deleted my miniforge installation
4. Forked again with main only branch.
5. Freshly installed miniforge from https://github.com/conda-forge/miniforge
 
And back again in terminal. No SUCCESS.
![Screenshot from 2024-05-05 10-49-58](https://github.com/catalyst-cooperative/pudl/assets/8761197/f62e1f0a-145e-4415-aa2c-f7e92e2a6266)
![Screenshot from 2024-05-05 10-50-18](https://github.com/catalyst-cooperative/pudl/assets/8761197/dd1dc486-cec2-43b4-9fe4-b43010199fbf)

![Screenshot from 2024-05-05 10-51-16](https://github.com/catalyst-cooperative/pudl/assets/8761197/a3a22257-a91f-4651-aed9-3ec2cb19a642)

**Reply by zaneselvans** [reply]
I'm so sorry you're having trouble with this.  I've just tried recreating my environment from scratch on the `main` branch locally and haven't had any issues, but I'm on MacOS so maybe it's a platform specific issue. This seems like a different kind of error than the one that @Nancy9ice was originally having with the mismatched package checksums.

With the `base` conda environment activated could you do `mamba list` to see what all packages and versions you've got installed there?  Maybe there's something different in there that's causing an issue.

Was there any additional output in that last stack trace? It doesn't seem like it gets to reporting its final error.  Or were they just the same as the ones above?

**Reply by dataengineergaurav** [reply]
I am using Ubuntu 22.04.4 LTS. I think it is the capture output error not the same as @Nancy9ice was facing earlier.

![image](https://github.com/catalyst-cooperative/pudl/assets/8761197/19c7bda1-027c-4a79-92c7-935d18db9a45)
![image](https://github.com/catalyst-cooperative/pudl/assets/8761197/a26c8f80-8a8e-4d4b-b8c1-834d174a8f4e)
![image](https://github.com/catalyst-cooperative/pudl/assets/8761197/4bbdb1ac-03ac-46a5-8a27-dd8f4c49ebe1)
![image](https://github.com/catalyst-cooperative/pudl/assets/8761197/007dd335-ca76-454e-bfd4-c9a8640d9543)

**Comment by dataengineergaurav** [comment]
I tried on WSL for Windows. It works now!
